### ISSUE TITLE
使用yolov8训练时，val/box_loss, val/cls_loss,val/dfl_loss值为nan

### ISSUE THREAD

**Issue by Xzzhurry** [ultralytics/ultralytics#9829]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

![1712450484376](https://github.com/ultralytics/ultralytics/assets/148302463/78098dd3-d5fe-45f5-a173-7903160c4ddb)

### Environment

Ultralytics YOLOv8.1.37 🚀 Python-3.11.8 torch-2.1.0+cu118 CUDA:0 (NVIDIA GeForce GTX 1660 Ti with Max-Q Design, 6144MiB)
Setup complete ✅ (12 CPUs, 23.8 GB RAM, 243.7/894.2 GB disk)

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@Xzzhurry hello, thanks for reaching out and for the detailed information! 👍

The occurrence of `nan` values in `val/box_loss`, `val/cls_loss`, and `val/dfl_loss` during training usually points towards issues with learning rate or data anomalies. Here are a few quick things you could try to resolve this:

1. **Adjust the Learning Rate:** Sometimes, a learning rate that's too high can cause instability. Try reducing it, for example:
    ```yaml
    lr0: 0.001  # smaller learning rate
    ```
    in your training YAML configuration.

2. **Check Data:** Ensure there are no corrupt or extremely unusual data points in your validation set.

3. **Gradient Clipping:** Introducing gradient clipping can prevent exploding gradients. Use the `--gradient_clip_val` argument when launching your training:
    ```bash
    yolo train data=mydata.yaml model=mymodel.yaml --gradient_clip_val 0.5
    ```
If after trying these steps the issue persists, it might be helpful to look into specific data preprocessing steps or review the distribution of your dataset. Let us know how it goes!

**Comment by Xzzhurry** [comment]
epochs=300 batch=4 lr0=0.01 resume=True workers=2  amp=False  这是我训练时设置的参数，lr我已经设置的比较低了

**Comment by Xzzhurry** [comment]
> 您好，感谢您伸出援手并提供详细信息！👍
> 
> 、 和训练期间出现的值通常指向学习率或数据异常问题。您可以尝试以下一些快速方法来解决此问题：`nan``val/box_loss``val/cls_loss``val/dfl_loss`
> 
> 1. **调整学习速度：**有时，过高的学习率会导致不稳定。尝试减少它，例如：
>    ```yaml
>    lr0: 0.001  # smaller learning rate
>    ```
>    
>    
>        
>          
>        
>    
>          
>        
>    
>        
>      
>    在训练 YAML 配置中。
> 2. **检查数据：**确保验证集中没有损坏或极不寻常的数据点。
> 3. **渐变剪切：**引入渐变剪切可以防止渐变爆炸。启动训练时使用以下参数：`--gradient_clip_val`
>    ```shell
>    yolo train data=mydata.yaml model=mymodel.yaml --gradient_clip_val 0.5
>    ```
> 
> 如果在尝试这些步骤后问题仍然存在，则查看特定的数据预处理步骤或查看数据集的分布可能会有所帮助。让我们知道它是怎么回事！

使用其他类型的机器训练我的数据没有问题，但在我自己的设备上我运行之后就会出现nan的问题。
![image](https://github.com/ultralytics/ultralytics/assets/148302463/e4ea4ef7-c85b-4b6d-895f-a724ddd04968)
这个是我的数据分布

**Comment by glenn-jocher** [comment]
@Xzzhurry 您好！感谢您的回复和提供额外信息 😊。您已尝试调整学习率且在其他机器上正常运行，表明您的设置基本合理。在您的设备上遇到的问题可能与环境配置或硬件兼容性有关。

建议尝试：
- **确保软件更新：**确保您的PyTorch、CUDA和其他相关库都是最新版本，以解决可能的兼容性问题。
- **温和地开始学习率：**尽管已经调整过，但考虑从一个非常小的学习率开始，例如0.0001，逐步增加看看是否有所改善。 
- **数据归一化：**确保所有输入数据都经过适当预处理和归一化，有时数据分布的小变化也可能导致模型训练问题。
- **确定硬件健康：**检查GPU和其他系统资源是否正常工作，有时硬件故障也会导致奇怪的行为。

希望这些建议能帮助到您！如果问题仍然存在，请随时分享更多信息，我们将尽力协助。
